### PR TITLE
Implement search field and options

### LINKED ISSUE #252: Implement search
**Task Details**

The following issue is to serve as a task list of issues pertaining to our **search**
Screen Shot 2019-08-01 at 3 37 13 AM

Subtasks
- [ ] **Issue #:** Implement search field
- [ ] **Issue #:** Implement search options menu

### PR DESCRIPTION
**Pull request by koderjoker**
**Description**
Adds user menu to the navigation bar.

Closes #252, #252 and #253

**Technical**
To view navbar
`npm install`
`npm install -g polymer-cli`
`cd topnav && polymer serve`

**Testing**
Click on search icon and compare behaviour as displayed in prototype (#247). Note that currently pressing the search button just opens and closes the search menu.

**Additions to be made later**
1. Close search menu when content is clicked
2. Adjustment of search icon
3. User menu to come after search options in tab order

### CODE REVIEW

File: topnav/search-menu.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import { LitElement, html, css } from 'lit-element'
```
Comment (iisa) [inline topnav/search-menu.js]: add semi-colon

File: topnav/search-menu.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    return {
+      searchMenuOpen: { type: Boolean },
+      searchMenuAnimate: { type: Boolean }
+    };
+  }
+
+  render() {
+    return html`
+    <nav
+      class="${this.searchMenuOpen ? 'search-menu open slide-in' : this.searchMenuAnimate ? 'search-menu slide-out': 'search-menu'}"
```
Comment (iisa) [inline topnav/search-menu.js]: Suggesgtion: make this a variable.

File: topnav/search-menu.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      aria-hidden="${this.searchMenuOpen ? 'false' : 'true'}"
```
Comment (iisa) [inline topnav/search-menu.js]: same with these conditionals

File: topnav/search-menu.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      aria-expanded="${this.searchMenuOpen ? 'true' : 'false'}"
+    >
+      <div class="search-options">
+        <div>
+          <input type="radio" id="metadata" name="search" value="metadata" checked>
```
Comment (iisa) [inline topnav/search-menu.js]: Suggestion: nest input in labels. remove `id` dependency

File: topnav/search-menu.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    `;
+  }
+
+  static get styles() {
+    return css`
+      .search-menu {
+        position: relative;
+        display: flex;
+        flex-direction: column;
+        margin: 0px;
```
Comment (iisa) [inline topnav/search-menu.js]: use `rem`. 1 rem = 10px

Comment (iisa) [reply topnav/search-menu.js]: or if 0px => 0

File: topnav/search-menu.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      .search-options {
+        align-self: center;
+      }
+      .search-options div {
+        padding: 8px;
+        color: #fff;
+        font-family: "Helvetica Neue";
+      }
+      .search-options a {
+        color: #428bca;
```
Comment (iisa) [inline topnav/search-menu.js]: Can we use css variables with these colors?

(cc: @shaneriley @jbuckner )

File: topnav/search-menu.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      .slide-out {
+        animation: slide-out 0.2s forwards;
+      }
+      .search-options {
+        align-self: center;
+      }
+      .search-options div {
+        padding: 8px;
+        color: #fff;
+        font-family: "Helvetica Neue";
```
Comment (iisa) [inline topnav/search-menu.js]: can we use variables for font?

File: topnav/user-menu.js [hunk 8]
```diff
@@ -31,40 +31,39 @@ class UserMenu extends LitElement {
...
         padding: 5px 10px;
-        transform: translate(0px, -500px);
+        transform: translate(0px, -1000px);
       }
       .open {
-        transform: translate(0px, 0px);
+        transform: translate(0px, -220px);
         z-index: 1;
       }
       @keyframes slide-in {
```
Comment (iisa) [inline topnav/user-menu.js]: Question: how can we reuse this animation? I see it in 2 places

File: topnav/search-menu.js (hunk 7 above)
Comment (jbuckner) [reply topnav/search-menu.js]: Yeah check out the [CSS Properties section](here: https://lit-element.polymer-project.org/guide/styles#css-properties). We can pass down CSS variables from a higher level. Same with the question above about color variables.

File: topnav/search-menu.js (hunk 5 above)
Comment (koderjoker) [reply topnav/search-menu.js]: Stalling till ratio is determined

**Pull request merged**